### ISSUE TITLE
Port all "simple" sharness tests

### ISSUE THREAD

**Issue by laurentsenta** [ipfs/gateway-conformance#26]
#3 relies on #22 and #10 which requires new features / larger tasks.

In parallel, we'll push more tests to the test suite to support bifrost

- [x] t0109-gateway-web-_redirects.sh (@lidel's note: afaik fixtures come from https://github.com/ipfs/specs/pull/290)
    - [x] basics: https://github.com/ipfs/gateway-conformance/pull/33
    - [x] dnslink: https://github.com/ipfs/gateway-conformance/pull/34
- [x] ~t0111-gateway-writeable.sh~  - no longer a thing (https://github.com/ipfs/kubo/pull/9743)
- [x] t0112-gateway-cors.sh - https://github.com/ipfs/gateway-conformance/pull/52
    - @lidel's note: Kubo returns multiple headers instead of array, tests should support both
- [x] t0113-gateway-symlink.sh
    - (@lidel's note: this ive very uncharted teritory, if anything weird is discovered here, comment on https://github.com/ipfs/specs/issues/368 :pray: )
    - #25 
- [x] t0114-gateway-subdomains.sh
    - [x] basics: https://github.com/ipfs/gateway-conformance/pull/9
    - [x] dnslink: https://github.com/ipfs/gateway-conformance/pull/35
    - [x] ipns + dnslink: https://github.com/ipfs/gateway-conformance/pull/53
- [x] t0115-gateway-dir-listing.sh - https://github.com/ipfs/gateway-conformance/pull/58
- [x] t0122-gateway-tar.sh #28 
- [x] t0123-gateway-json-cbor.sh
    - [x] non-ipns part - #25
    - [x] ipns part - https://github.com/ipfs/gateway-conformance/pull/59

Other tests implemented along other tickets:

- [x] ~t0117-gateway-block.sh~ #3
- [x] ~t0118-gateway-car.sh~ #3
- [x] ~t0124-gateway-ipns-record.sh~ #3
- [x] ~t0400-api-no-gateway.sh~ - kubo specific

## follow-up tasks

- [x] t0116-gateway-cache.sh  (@lidel's note: pretty important one) https://github.com/ipfs/gateway-conformance/pull/97

**Comment by galargh** [comment]
@laurentsenta I'm going to try to pick up `t0116-gateway-cache.sh`. I'll report at the end of the day what the status is.

**Comment by BigLep** [comment]
@laurentsenta : is the checklist accurate of what's remaining?  There look to be unchecked things that are merged and https://github.com/ipfs/gateway-conformance/pull/30#issuecomment-1586783446 was unclear to me whether that PR is going to be merged.

I am asking because I'm trying to get a sense of what's remaining and where others need to help review so we can close this.  Thanks.

**Comment by laurentsenta** [comment]
@BigLep the ticket is up to date now, we'll tackle 116 in a follow-up task, I really want to move on with all of these.

We were missing one last pass to double-check that everything was ported correctly, and remove the "wip" comments,
PR: https://github.com/ipfs/gateway-conformance/pull/92
